File: src/WebUI/libs/projects/data-access/src/lib/models/get-project-data.response.ts
import { GridPanelModel, GridStringModel, PanelLinkModel } from '@shared/data-access/models'

export interface GetProjectDataResponse {
  id: string
  name: string
  createdTime: string
  createdById: string
  lastModifiedTime: string
  panels: GridPanelModel[]
  panelLinks: PanelLinkModel[]
  strings: GridStringModel[]
}

/*

 public string Id { get; set; } = default!;
 public DateTime CreatedTime { get; set; }
 public DateTime LastModifiedTime { get; set; }*/

File: src/Services/Projects/Projects.Contracts/Data/GetProjectDataResponse.cs
namespace Projects.Contracts.Data;

public class GetProjectDataResponse : IProjectItemDto
{
    public string Name { get; set; } = default!;
    public IEnumerable<StringDto> Strings { get; set; } = default!;
    public IEnumerable<PanelDto> Panels { get; set; } = default!;
    public IEnumerable<PanelLinkDto> PanelLinks { get; set; } = default!;
    public IEnumerable<PanelConfigDto> PanelConfigs { get; set; } = default!;
    public string Id { get; set; } = default!;
    public DateTime CreatedTime { get; set; }
    public DateTime LastModifiedTime { get; set; }
    public string CreatedById { get; set; } = default!;
}